node/avn-service/src/lib.rs

```diff
@@ -428,16 +421,15 @@ where
             let secp = Secp256k1::new();
             let keystore_path = &req.state().keystore_path;
 
-            let data_to_sign: Vec<u8> =
-                hex::decode(req.param("data_to_sign")?.trim_start_matches("0x")).map_err(|e| {
+            let data_to_sign = req.param("data_to_sign")?;
+            let hashed_message =
+                hash_with_ethereum_prefix(&data_to_sign.to_string()).map_err(|e| {
```

Comment:
what happened with the `trim`?

Reply:
its happening in the `hash_with_ethereum_prefix` function